### PR TITLE
fix: lys congrats refetch site cache status

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR addresses the feedback in the [previous PR](https://github.com/woocommerce/woocommerce/pull/48586) about cached content status not being correctly refetched on page refresh or direct url access.

It also refactors the content page and add tests for it.

Details:
- Added the site cached content check to the main content state machine
- Moved side effects out of the Success page components so that the component can be easily tested
- Moved tracks out of the Success page components so that they are colocated with the side effects
- Made use of the loading state in the state machine instead of the temporary one that was there before
- Moved the Survey bit out into its own encapsulated component since its inner functioning is independent of the Congrats screen
- Tests for the Survey

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

image

These testing instructions are mostly from https://github.com/woocommerce/woocommerce/pull/46103

#### Test success page
1. Create a new JN site with this branch.
3. Enable `launch-your-store` feature flag.
4. Go to WooCommerce > Home > Launch Your Store
2. Click on `Launch Store` and `Launch Store Success`
3. Observe confetti animation after the page is loaded

#### Test cache status
1. Create a new JN site with this branch.
2. Go to https://your-jn-site/wp-admin/plugin-editor.php
3. Choose `src/Internal/ComingSoon/ComingSoonRequestHandler.php`
4. Return `true` from `should_show_coming_soon` function for testing. 
5. Go to `WooCommerce -> Home` and click on `Launch your store` task
6. Click on `Launch your store` button from the side bar.
7. You should see the copy for the cached site.
8. Refreshing it should still show the cached site copy

#### Test copy link
1. Click on `Copy link`
2. Observe the copy button is changed to `Copied!` and then revert after a couple of seconds
3. Paste anywhere and ensure the link is the site's homepage

#### Test visit store link
1. Click on `Visit your store`
2. Observe it opens up the homepage in a new tab

#### Test survey
1. Click on any emoji next to `How was the experience of launching your store?`
2. Observe a textarea labelled `Why do you feel that way?` is shown
3. Type anything into the textarea
4. Click `Cancel`
5. Observe the textarea disappear and emoji is deselected
6. Click on any emoji again
7. Observe the textarea is brought back with previous contents
9. Click on `Send`
10. Observe `🙌 We appreciate your feedback!` is shown
11. Refresh the page
12. Observe survey form (emoji list) is no longer shown

#### Test next steps
1. With no tasks completed, observe the next steps are `Promote your products`, `Provide more ways to pay`, and `Power up your store`
2. Click on `Promote products` on a new tab
3. Observe you're navigated to Marketing screen
4. Install `TikTok for WooCommerce` plugin
5. Refresh success screen
6. Observe the new list is `Provide more ways to pay`, `Power up your store`, and `Build customer relationships`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/launch-your-store/hub/main-content/xstate.tsx [hunk 1]
```diff
@@ -59,6 +63,31 @@ export const mainContentMachine = setup( {
 		assignSiteCachedStatus: assign( {
 			siteIsShowingCachedContent: true,
 		} ),
+		recordSurveyResults: ( { event } ) => {
+			assertEvent( event, 'COMPLETE_SURVEY' );
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/launch-your-store/hub/main-content/xstate.tsx:67]: Just curious, why do we need assertEvent here?

**Review by ilyasfoo — APPROVED** [review]
Nice work! Thanks for adding tests for Survey component too. LGTM! 🚢

**Review by psealock — APPROVED** [review]
Nice work here @rjchow, this is testing well and the code change look good too. Thanks for the added tests 🚀

File: plugins/woocommerce-admin/client/launch-your-store/hub/main-content/xstate.tsx (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/launch-your-store/hub/main-content/xstate.tsx:67]: XState V5 doesn't narrow down the type of the event to the expected event, so without assertEvent you need to handle all possible event types in the function :| (i.e it's here as a type helper)

**Pull request merged**